[
  {
    "project": "fieldOfView/Cura-OctoPrintPlugin",
    "commit": "b89e2303929c361b2381d11592eb3c473b47241b",
    "filename": "NetworkReplyTimeout.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/fieldOfView-Cura-OctoPrintPlugin/NetworkReplyTimeout.py",
    "file_hunks_size": 1,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "NetworkReplyTimeout.py:18:69 Invalid type [31]: Expression `typing.Optional[typing.Callable[PyQt5.QtNetwork.QNetworkReply]]` is not a valid type.",
    "message": " Expression `typing.Optional[typing.Callable[PyQt5.QtNetwork.QNetworkReply]]` is not a valid type.",
    "rule_id": "Invalid type [31]",
    "warning_line_no": 18,
    "warning_line": "    def __init__(self, reply: QNetworkReply, timeout: int, callback: Optional[Callable[QNetworkReply]] = None) -> None:"
  }
]